### DISCUSSION TITLE
Add Keys() and Values() for dict-type vars ?

### DISCUSSION THREAD

**Discussion by yellowbean** [reflex-dev/reflex#3027] [Ideas]
## Background

There are couple useful operators for vars

https://reflex.dev/docs/vars/var-operations/

Since it's quite common to have `dict` type as vars , `  myDict: dict[str,int]` 

do you think it would be great to have Keys() or Values() on the `myDict` ?

then equivalent to 

    State(rx.State):
       myDict: dict[str, str]

       @rx.var
       def myDictKeys():
             return list(myDict.keys())

If there is a lot of vars like `myDict` , then the code just will have unnecessary lines.

## Expected behavior 

In the component

    # build all keys to text component.
    rx.foreach( State.myDict.Keys(), rx.text)

**Comment by masenf** [comment]
Currently you can do this with normal foreach and a lambda:

```python
class State(rx.State):
    my_d: dict[str, str] = {"a": "1", "b": "2", "c": "3"}


def index() -> rx.Component:
    return rx.vstack(
        rx.foreach(
            State.my_d,
            lambda kv: rx.hstack(
                rx.heading(kv[0])
            ),
        )
    )
```

But i agree, it might be more clear to have `.keys()` or `.values()` defined as a Var operation, so that you don't have to deal with the tuple.
